### PR TITLE
Allow build script to detect more recent Windows 10 SDK versions

### PR DESCRIPTION
**Pull request by PetSerAl**
## PR Summary

PowerShell native windows binaries can be compiled using recent Window 10 SDK versions (for example 10.0.17134.0), but build script `build.psm1` does not detect them installed. This PR allow build script to detect them.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by daxian-dbw — APPROVED** [review]

File: build.psm1 [hunk 1]
```diff
@@ -170,7 +170,8 @@ function Test-Win10SDK {
     #
     # A slightly more robust check is for the mc.exe binary within that directory.
     # It is only present if the SDK is installed.
-    return (Test-Path "${env:ProgramFiles(x86)}\Windows Kits\10\bin\x64\mc.exe")
+    return (Test-Path "${env:ProgramFiles(x86)}\Windows Kits\10\bin\x64\mc.exe") -or
+           (Test-Path "${env:ProgramFiles(x86)}\Windows Kits\10\bin\10.0.*.0\x64\mc.exe")
```
Comment (iSazonov) [inline build.psm1]: I see 5 folder on my computer and I don't know path default...
Why we test full path here and use env path search in line 236?
Maybe test `Get-Command -Application mc.exe`?

Comment (PetSerAl) [reply build.psm1]: >Why we test full path here and use env path search in line 236?

Line 236? You mean this?:
https://github.com/PowerShell/PowerShell/blob/c43ae6a110ec1c3555276d14e7b7b9bab3fa2487/build.psm1#L236

>Maybe test `Get-Command -Application mc.exe`?

`mc.exe` is not in `PATH` by default. It is `vcvarsall.bat` responsibility to locate installed SDK and add it to path:
https://github.com/PowerShell/PowerShell/blob/c43ae6a110ec1c3555276d14e7b7b9bab3fa2487/build.psm1#L265

Comment (iSazonov) [reply build.psm1]: Sorry the line I meant
https://github.com/PetSerAl/PowerShell/blob/8bee005699178008ca78efa306b4d196c53c0a54/build.psm1#L266

On my computer the exe is in PATH
```
where.exe mc.exe
C:\Program Files (x86)\Microsoft SDKs\Windows\v7.1A\Bin\MC.Exe
```
Perhaps new installer doesn't add this in PATH.

Comment (PetSerAl) [reply build.psm1]: For me it is not in `PATH`, unless you call `vcvarsall.bat` in initialize developer environment:
```cmd
C:\>where.exe mc.exe
INFO: Could not find files for the given pattern(s).

C:\>"\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvarsall.bat" x64
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

C:\>where.exe mc.exe
C:\Program Files (x86)\Windows Kits\10\bin\10.0.17134.0\x64\mc.exe
```

Comment (iSazonov) [reply build.psm1]: Therefore, we can not test the existence of mc.exe before launching `vcvarsall.bat`, is it?

**Comment by TravisEz13** [comment]
This is only needed for native code?  @adityapatwardhan   should this be moved to the PowerShell-Native repo?

**Comment by adityapatwardhan** [comment]
Yes the changes should be made in the https://github.com/PowerShell/PowerShell-native. Please open a PR over there. Thanks!

**Comment by daxian-dbw** [comment]
@adityapatwardhan I already merged #7218. Just want you to know in case any changes to native build script should happen in the other repo.

**Comment by daxian-dbw** [comment]
@adityapatwardhan I think it's probably better to keep the native-build script up-to-date in the powershell repo until we completely remove it.

**Comment by iSazonov** [comment]
What is the cut-off date?

**Comment by daxian-dbw** [comment]
@iSazonov I don't know a hard cut-off day, but I guess we want to get it done before GA.

File: build.psm1 (hunk 1 above)
Comment (PetSerAl) [reply build.psm1]: As far as I understand, yes. But I am not the one who implement original check, so I can not state exact reasons, why it was implemented this way.

**Comment by daxian-dbw** [comment]
I will submit a PR in the powershell-native repo for this change and merge and then merge this PR, so that both side are in sync regarding this change.

**Review by TravisEz13 — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@PetSerAl Thank you for submitting this PR. The current code is PowerShell/PowerShell-Native repo already has this fix. Closing this PR.

**Pull request closed without merging**